### PR TITLE
feat: XDSMarkdown — zero-dep markdown renderer with incremental parser

### PR DESCRIPTION
**Pull request by cixzhang**
## Summary

Zero-dependency markdown renderer that lives in `@xds/core`. Custom parser produces a typed AST, renderer maps it directly to XDS components.

### Why custom parser?
- Zero dependencies — no `markdown-to-jsx`, no `react-markdown`
- Direct XDS component output — `XDSCodeBlock` for fenced code, `XDSCode` for inline code
- Full type safety — typed AST nodes, no `any` casts
- Incremental streaming parser — caches settled blocks, 3-4x faster for long responses
- Lives in `@xds/core` — no separate package needed

### Parser (~280 lines)
**Block:** headings, paragraphs, fenced code blocks, blockquotes, ordered/unordered/task lists, GFM tables with alignment, horizontal rules, images
**Inline:** bold, italic, strikethrough, inline code, links, images, escape sequences

### Incremental streaming
Tracks settled vs unsettled blocks. Only re-parses the current incomplete block on each token. Code fences prevent false settlement across blank lines.

### Component props
- `children`: markdown string
- `density`: `'default'` | `'compact'`
- `maxHeadingLevel`: clamp heading levels
- `isStreaming`: incremental parse + blinking cursor
- `onLinkClick`: navigation prevention callback

### Tests: 73 passing
- 24 parser correctness tests
- 13 incremental streaming edge cases
- 10 performance benchmarks
- 26 component render tests

---

### CODE REVIEW

File: packages/core/src/Markdown/XDSMarkdown.tsx [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+      const handleClick = onLinkClick
+        ? (e: React.MouseEvent<HTMLAnchorElement>) => {
+            const result = onLinkClick(node.href, e);
+            if (result === false) {
+              e.preventDefault();
+            }
+          }
+        : undefined;
+      return (
+        <a
```
Comment (cixzhang) [inline packages/core/src/Markdown/XDSMarkdown.tsx:327]: Links should use XDSLink or an element from the link provider

File: packages/core/src/Markdown/XDSMarkdown.tsx [hunk 2]
```diff
@@ -0,0 +1,423 @@
...
+          ))}
+        </Tag>
+      );
+    }
+    case 'table': {
+      const alignStyle = (a: typeof node.alignments[number]) =>
+        a === 'center' ? styles.alignCenter : a === 'right' ? styles.alignRight : styles.alignLeft;
+      return (
+        <div key={index} {...stylex.props(blockSpacing)}>
+          <table {...stylex.props(styles.table)}>
```
Comment (cixzhang) [inline packages/core/src/Markdown/XDSMarkdown.tsx:549]: Tables should use XDSTable and we could embed some client side features like sorting, resize, etc.

File: packages/core/src/Markdown/XDSMarkdown.tsx [hunk 3]
```diff
@@ -0,0 +1,423 @@
...
+                  ))}
+                </tr>
+              ))}
+            </tbody>
+          </table>
+        </div>
+      );
+    }
+    case 'hr':
+      return <hr key={index} {...stylex.props(styles.hr, blockSpacing)} />;
```
Comment (cixzhang) [inline packages/core/src/Markdown/XDSMarkdown.tsx]: Should use XDSDivider

File: packages/core/src/Markdown/XDSMarkdown.tsx [hunk 4]
```diff
@@ -0,0 +1,423 @@
...
+    case 'image':
+      return (
+        <p key={index} {...stylex.props(blockSpacing)}>
```
Comment (cixzhang) [inline packages/core/src/Markdown/XDSMarkdown.tsx]: Hmm perhaps this should use XDSAspectRatio for consistency and corner radii.

File: packages/core/src/Markdown/XDSMarkdown.tsx [hunk 5]
```diff
@@ -0,0 +1,423 @@
...
+          onClick={handleClick}
+          {...(isExternal ? {target: '_blank', rel: 'noopener noreferrer'} : {})}
+          {...stylex.props(styles.link)}
+        >
+          {node.children.map((c, i) => renderInline(c, i, onLinkClick))}
+        </a>
+      );
+    }
+    case 'image':
+      return <img key={index} src={node.src} alt={node.alt} {...stylex.props(styles.image)} />;
```
Comment (cixzhang) [inline packages/core/src/Markdown/XDSMarkdown.tsx]: XDSAspectRatio

File: packages/core/src/Markdown/XDSMarkdown.tsx [hunk 6]
```diff
@@ -0,0 +1,423 @@
...
+    case 'blockquote':
+      return (
+        <blockquote key={index} {...stylex.props(styles.blockquote, blockSpacing)}>
+          {node.children.map((c, i) => renderBlock(c, i, density, maxHeadingLevel, onLinkClick))}
+        </blockquote>
+      );
+    case 'list': {
+      const Tag = node.ordered ? 'ol' : 'ul';
+      return (
+        <Tag
```
Comment (cixzhang) [inline packages/core/src/Markdown/XDSMarkdown.tsx:509]: We can use XDSList with list styles

File: packages/core/src/Markdown/XDSMarkdown.tsx [hunk 7]
```diff
@@ -0,0 +1,423 @@
...
+          {...(node.ordered && node.start != null ? {start: node.start} : {})}
+          {...stylex.props(styles.list, blockSpacing)}
+        >
+          {node.items.map((item, i) => (
+            <li
+              key={i}
+              {...(item.checked != null ? stylex.props(styles.taskItem) : {})}
+            >
+              {item.checked != null && (
+                <input
```
Comment (cixzhang) [inline packages/core/src/Markdown/XDSMarkdown.tsx]: Use XDSChexboxInput
